use PKG_CHECK_VAR for defining WAYLAND_PROTOCOLS_DATADIR

This allows to override the wayland-protocols pkgdatadir with the
WAYLAND_PROTOCOLS_DATADIR from environment.

pkgconfig would return an absolute path in /usr/share/wayland-protocols
for the pkgdatadir value, which is not suitable for cross-compiling.

Signed-off-by: Khem Raj <raj.khem@gmail.com>
Upstream-Status: Pending

Index: mesa-18.1.2/configure.ac
===================================================================
--- mesa-18.1.2.orig/configure.ac
+++ mesa-18.1.2/configure.ac
@@ -1808,7 +1808,7 @@ for plat in $platforms; do
         PKG_CHECK_MODULES([WAYLAND_CLIENT], [wayland-client >= $WAYLAND_REQUIRED])
         PKG_CHECK_MODULES([WAYLAND_SERVER], [wayland-server >= $WAYLAND_REQUIRED])
         PKG_CHECK_MODULES([WAYLAND_PROTOCOLS], [wayland-protocols >= $WAYLAND_PROTOCOLS_REQUIRED])
-        WAYLAND_PROTOCOLS_DATADIR=`$PKG_CONFIG --variable=pkgdatadir wayland-protocols`
+        PKG_CHECK_VAR([WAYLAND_PROTOCOLS_DATADIR], [wayland-protocols >= $WAYLAND_PROTOCOLS_REQUIRED], pkgdatadir)
 
         AC_PATH_PROG([WAYLAND_SCANNER], [wayland-scanner], [:])
 
